### PR TITLE
Add from-job and from-pr option to breeze ci-image load

### LINKED ISSUE #45269: Add "load" command flags to restore CI image from Job ID / PR ID
With https://github.com/apache/airflow/pull/45261 we switched to storing "per-job" image dumps as artifacts in the jobs from GitHub registry - this also means that we cannot use Commit SHA TAG to pull the image from registry when we want to reproduce a CI job environment.

For now we can do it manually - by downloading the artifact and using `breeze ci-image load --image-file <file>`  - but this can be all easily automated.

The `load` command could have optional `--from-job <JOB>` and `--from-pr PR` flags that could use GiHub API to find and download the right artifact before loading. For JOB - it would take it from the JOB, from PR it would load it from the latest JOB for the PR. 

We would also add diagnostic to error out if the image is not with the right platform (for now only `linux/amd` is supported) and in the future when we add ARM support via ARC (https://github.com/apache/airflow/issues/44512) - we should be able to also produce and store ARM image as artifact, and in this case we would just find and use the right one depending on the platform we are in.

### PR DESCRIPTION
**Pull request by gopidesupavan**
closes: #45269 

follow up from #45266

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: dev/breeze/src/airflow_breeze/utils/github.py [hunk 1]
```diff
@@ -177,3 +180,123 @@ def get_tag_date(tag: str) -> str | None:
...
+    with zipfile.ZipFile(temp_file, "r") as zip_ref:
+        zip_ref.extractall("/tmp/")
+
+    os.remove(temp_file)
+
+
+def download_artifact_from_pr(pr: str, output_file: str):
+    import requests
+
+    pr_url = f"https://api.github.com/repos/apache/airflow/pulls/{pr}"
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/utils/github.py]: Nit: we should also support #PR form - similarly as `gh checkout pr`does

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Nice! One small nit

File: dev/breeze/src/airflow_breeze/utils/github.py (hunk 1 above)
Comment (gopidesupavan) [reply dev/breeze/src/airflow_breeze/utils/github.py]: Sorry didnt get :), this method is for pr from, it needs pr number. we can provide `--from-pr 123`

Comment (gopidesupavan) [reply dev/breeze/src/airflow_breeze/utils/github.py]: You mean pull request number with hash? like `--from-pr #45287`?

Comment (gopidesupavan) [reply dev/breeze/src/airflow_breeze/utils/github.py]: got it resolved :)

**Pull request merged**